src/rpc.rs

```diff
@@ -838,7 +838,7 @@ pub struct RpcErrorOwned {
     code: i64,
     message: String,
     #[serde(skip_serializing_if = "Option::is_none")]
-    data: Option<Box<RawValue>>,
+    data: Option<serde_json::Value>,
```

Comment:
I don't see how it fixes the problem, do you have a proof it does?

Reply:
And what was the problem in the first place

Reply:
```rust
use serde::Deserialize;
use serde_json::value::RawValue;

fn main() {
    let ok = r#"{"jsonrpc":"2.0","result": "ok","id":1}"#;
    let error = r#"{
        "jsonrpc": "2.0",
        "error": {
        "code": -32005,
        "message": "Node is behind by 42 slots",
        "data": {
          "numSlotsBehind": 42
        }
    },
        "id": 1
    }"#;

    let ok: Flatten<Response<String>> = serde_json::from_str(ok).unwrap();
    let err: Flatten<Response<String>> = serde_json::from_str(error).unwrap();
    let not_parsed = serde_json::from_str::<Flatten<ResponseBroken<String>>>(error);

    println!("OK:{:?} \nERR: {:?}", ok, err);

    assert!(not_parsed.is_err());
}

#[derive(Deserialize, Debug)]
#[serde(rename_all = "lowercase")]
enum Response<T> {
    Result(T),
    Error(ResponseError),
}

#[derive(Deserialize, Debug)]
#[serde(rename_all = "lowercase")]
enum ResponseBroken<T> {
    Result(T),
    Error(ResponseErrorBroken),
}

#[derive(Deserialize, Debug)]
struct ResponseError {
    code: i64,
    message: String,
    data: Option<serde_json::Value>,
}

#[derive(Deserialize, Debug)]
struct ResponseErrorBroken {
    code: i64,
    message: String,
    data: Option<Box<RawValue>>,
}

#[derive(Deserialize, Debug)]
struct Flatten<T> {
    #[serde(flatten)]
    inner: T,
}
```

Here's the proof, and the error is the same as the one that was reported. The problem it seems that when we have a field with RawValue, and flatten is involved, serde fails to deserialize it. I've found [one bug report](https://github.com/serde-rs/json/issues/599) very similar to ours, but no generic workaround for the problem, using value turned out to work though.

Reply:
ok